TypeScript definitions

borekb opened the pull request:
Provides a bit nicer developer experience:

![image](https://user-images.githubusercontent.com/101152/30585002-c1cb0390-9d2b-11e7-9a30-9b26cd8f9f72.png)

wmhilton commented:
I like it! Quick question though. Technically the code still works if signal is a number (because process.kill allows signal [to be a number](https://nodejs.org/api/process.html#process_process_kill_pid_signal)) and if pid is a string (because we manually convert pid to an integer because of [this bug in an old version of Node](https://github.com/pkrumins/node-tree-kill/blob/master/index.js#L68)). Should we include those possibilities in the typescript definition or leave them as-is to encourage people to use the expected type?

borekb commented:
You're right about the string / number for `signal`, I've updated the type definitions in https://github.com/pkrumins/node-tree-kill/pull/19/commits/13752aba4810c0ff567d3c8ee7c4e8dd5e9a1fc5 and done some other smaller improvements. `pid` always seems to be a number, at least in reasonably recent versions of Node, so I kept it as `number` only (feel free to update it if you think it's incorrect).

I've also added test/types.ts as a playground, works nicely in VSCode (and in Atom I would assume).

wmhilton commented:
Very nice! Published in `v1.2.0`
